src/interactive-window/interactiveWindow.ts

```diff
@@ -548,14 +617,12 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         isDebug: boolean
     ) {
         traceInfoIfCI('InteractiveWindow.ts.createExecutionPromise.start');
-        const [kernel, { cell, wasScrolled }, editor] = await Promise.all([
-            this._kernelPromise.promise,
-            notebookCellPromise,
-            this._editorReadyPromise
-        ]);
-        if (!kernel) {
-            return false;
-        }
+        // Wait for the cell to get created (required to insert error messages into cell output).
+        // When we wait for kernel to start, its possible it will fail, and
+        // higher up, we insert errors into the cell (but the cell needs to have been inserted).
+        const { cell, wasScrolled } = await notebookCellPromise;
```

Comment:
Not a fan of this, personally, I think we should just add the error messages into the cell right here.
Its weird that CodeWatcher has a method `addCodeCell`, and that handles the errors and the calls into this same class to update the cell with the errors.

I agree error handling should be done higher up, but the way the code is written right now:
* CodeWatcher doesn't handle errors related to ipykernel not being found (that internal block displays messages & handles those errors)
* CodeWatcher handles errors in cells and then calls back into IW to add error messages

Feels all too weird right now, hence this hacky code, its very very easy for this to break (will need a test to ensure we don't break this), to break this all we need to do is use a `Promise.all` as follows (which is the old code):
```typescript
const [kernel, { cell, wasScrolled }, editor] = await Promise.all([
            this._kernelPromise.promise,
            notebookCellPromise,
            this._editorReadyPromise
        ]);
```

Reply:
There are two options:
* Refactor
* Write a test to ensure we don't break hack

Reply:
I prefer the former
Looking at code watcher, we don't do anything special with errors, all we do is handle the error and display it back in the IW.
Moving that into this part will make all of that simpler (IMHO)

Reply:
Sounds good to me. Don't have the codewatcher handle errors. Doesn't seem like it should as how is it to know how the interactive window should display them?

Reply:
We have two methods today in `IDataScienceErrorHandler`, `handleError` and `getErrorMessageForDisplayInCell` 
Yes, the code watcher handles errors, and my suggestion is to :
* Let the code watcher handle the errors by calling `IDataScienceErrorHandler.handleError`
* Let the IW display errors in the IW cells by calling `IDataScienceErrorHandler.getErrorMessageForDisplayInCell`  & display that message in the cells.

This way we remove the call backs, and the like (call back to get the cell information to insert errors).
However we now need to pass the `leftCount` all the way into the IW, so that it can change the error message based on whether there are any more pending cells..

Reply:
I think the IW could tell if there are pending cells because they should be in the list of cells in the IW now that the cells are queued up before execution.

So you might need the leftCount.

Reply:
Fixed that issue in this PR https://github.com/microsoft/vscode-jupyter/pull/9483, yes we don't need that anymore.